src/algebra/lie/direct_sum.lean

```diff
@@ -87,27 +89,121 @@ instance : lie_ring (⨁ i, L i) :=
 @[simp] lemma bracket_apply (x y : ⨁ i, L i) (i : ι) :
   ⁅x, y⁆ i = ⁅x i, y i⁆ := zip_with_apply _ _ x y i
 
-instance : lie_algebra R (⨁ i, L i) :=
+instance lie_algebra : lie_algebra R (⨁ i, L i) :=
 { lie_smul := λ c x y, by { ext, simp only [
     zip_with_apply, smul_apply, bracket_apply, lie_smul] },
   ..(infer_instance : module R _) }
 
 variables (R ι L)
 
 /-- The inclusion of each component into the direct sum as morphism of Lie algebras. -/
-def lie_algebra_of [decidable_eq ι] (j : ι) : L j →ₗ⁅R⁆ ⨁ i, L i :=
-{ map_lie' := λ x y, by
+@[simps] def lie_algebra_of [decidable_eq ι] (j : ι) : L j →ₗ⁅R⁆ ⨁ i, L i :=
+{ to_fun   := lof R ι L j,
```

Comment:
It might make sense to have this unfold to `of` instead of `lof`

Reply:
I presume you mean using `to_fun   := of L j` instead of `to_fun   := lof R ι L j`?

I guess the advantage is that `lie_algebra_of_apply` will then drop `R`, allowing consumers to forget `R` sooner. OK I've talked myself into this by these words, I'll give it a go!

Reply:
Yes, exactly

Reply:
OK, done!